nodes/filter/nodes.go

```diff
@@ -0,0 +1,225 @@
+	return &OnlyBetween{body, 0}, nil
+}
+
+func NewOnlyGreater(body *node.Spec) (node.Node, error) {
+	body = node.Defaults(body, onlyGreater, category)
+	inputs := node.BuildInputs(node.BuildInput(node.In, node.TypeFloat, nil, body.Inputs), node.BuildInput(node.InMin, node.TypeFloat, nil, body.Inputs))
+	outputs := node.BuildOutputs(node.BuildOutput(node.Out, node.TypeFloat, nil, body.Outputs))
+	body = node.BuildNode(body, inputs, outputs, nil)
+	body.SetHelp(fmt.Sprintln("This node filters ‘input’ values. Only Numeric ‘input’ values greater than ‘OutMax’ are passed to ‘output’. Output the last known input float value if the input is less than 'OutMax'. Out put nil if any of the inputs are nil."))
+	return &OnlyGreater{body, 0}, nil
+}
+
+func NewOnlyLower(body *node.Spec) (node.Node, error) {
+	body = node.Defaults(body, onlyLower, category)
+	inputs := node.BuildInputs(node.BuildInput(node.In, node.TypeFloat, nil, body.Inputs), node.BuildInput(node.InMax, node.TypeFloat, nil, body.Inputs))
+	outputs := node.BuildOutputs(node.BuildOutput(node.Out, node.TypeFloat, nil, body.Outputs))
+	body = node.BuildNode(body, inputs, outputs, nil)
+	body.SetHelp(fmt.Sprintln("This node filters ‘input’ values. Only Numeric ‘input’ values less than 'InMax' are passed to ‘output’. Output the last known input float value if the input is greater than 'OutMin'. Out put nil if any of the inputs are nil."))
+	return &OnlyLower{body, 0}, nil
+}
+
+func NewPreventDuplicates(body *node.Spec) (node.Node, error) {
+	body = node.Defaults(body, preventDuplicates, category)
+	inputs := node.BuildInputs(node.BuildInput(node.In, node.TypeFloat, nil, body.Inputs))
+	outputs := node.BuildOutputs(node.BuildOutput(node.Out, node.TypeFloat, nil, body.Outputs))
+	body = node.BuildNode(body, inputs, outputs, nil)
+	body.SetHelp(fmt.Sprintln("This node filters ‘input’ values. All ‘input’ values are passed to ‘output’ EXCEPT ‘input’ values which are equal to the previous ‘input’ value. Output the last known input float value if the subsequent input value is the same. Out put nil if the input is nil."))
+	return &PreventDuplicates{body, 0}, nil
+}
+
+// preventNull, output the last known value if the input became null.
+// lastValue is defaulted to be 0
+func (inst *PreventNull) Process() {
+	input, null := inst.ReadPinAsFloat(node.In)
+
+	if null {
+		inst.WritePinFloat(node.Out, inst.lastValue)
+	} else {
+		inst.WritePinFloat(node.Out, input)
+		inst.lastValue = input
+	}
+}
+
+// preventEqualFloat, output the last known value if two inputs equal to each other. Out put nil if any of the inputs are nil.
+// lastValue is defaulted to be 0
+func (inst *PreventEqualFloat) Process() {
+	input, inputNull := inst.ReadPinAsFloat(node.In)
+	match, matchNull := inst.ReadPinAsFloat(node.InputMatch)
+
+	if inputNull || matchNull {
```

Comment:
this logic is incorrect.  The only comparison we need with `null` is if the `match` value is `null` (ie. functions the same as the `prevent-null` node.  So => if `!matchNull` AND `inputNull` then it's ok to set the output to `null`.